### ISSUE TITLE
Backport lazy loading of data module

### ISSUE THREAD

**Issue by effigies** [scikit-image/scikit-image#6069]
## Description

The obligatory loading of `skimage.data` and downloading images (under certain circumstances) can make scikit-image a finicky 
dependency. #5996 has introduced lazy loading for v0.19.x, and this would be nice to have this in a released version soon. 

The specific issue is that, if the conda-forge package is installed, then any import of `scikit-image` will attempt to download images. We've run into a case (nipreps/fmriprep#2644) where this runs into a "No space left on device" error. I reproduce below with a permission error.

Installing the same wheel from PyPI does not produce this issue, since the data is found in the package directory. I don't know if this should just be an issue on the conda-forge feedstock, but it seems that a release that includes #5996 would make a fix there moot.

## Way to reproduce

This occurs specifically for a conda-forge installation:
```bash
$ conda create -n skimage-bug -c conda-forge -c anaconda -y python=3.9 scikit-image=0.18
$ conda activate skimage-bug
```

We need a cache directory that cannot be written:

```bash
$ rm -r $HOME/.cache/scikit-image || true
$ install -d -m 000 $HOME/.cache/scikit-image # Pre-create cache directory with no permissions
```

```Python
>>> from skimage.morphology import ball
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/chris/miniconda3/envs/skimage-bug/lib/python3.9/site-packages/skimage/__init__.py", line 135, in <module>
    from .data import data_dir
  File "/home/chris/miniconda3/envs/skimage-bug/lib/python3.9/site-packages/skimage/data/__init__.py", line 270, in <module>
    _init_pooch()
  File "/home/chris/miniconda3/envs/skimage-bug/lib/python3.9/site-packages/skimage/data/__init__.py", line 255, in _init_pooch
    os.makedirs(data_dir, exist_ok=True)
  File "/home/chris/miniconda3/envs/skimage-bug/lib/python3.9/os.py", line 215, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/home/chris/miniconda3/envs/skimage-bug/lib/python3.9/os.py", line 225, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/home/chris/.cache/scikit-image/0.18.3'
```

## Version information
```python
# Paste the output of the following python commands
from __future__ import print_function
import sys; print(sys.version)
import platform; print(platform.platform())
import skimage; print(f'scikit-image version: {skimage.__version__}')
import numpy; print(f'numpy version: {numpy.__version__}')
```

```python
3.9.7 | packaged by conda-forge | (default, Sep 29 2021, 19:20:46) 
[GCC 9.4.0]
Linux-5.11.0-40-generic-x86_64-with-glibc2.31
scikit-image version: 0.18.3
numpy version: 1.21.4
```

**Comment by hmaarrfk** [comment]
We just released 0.19.0.

Packages are being built at conda-forge.
https://github.com/conda-forge/scikit-image-feedstock/pull/85

Thank you for your patience.

**Comment by effigies** [comment]
Thanks!
